options.go

```diff
@@ -41,6 +41,14 @@ func Resolve(options interface{}, flagSet *flag.FlagSet, cfg map[string]interfac
 		//    deprecated - (optional) the name of the deprecated command line flag
 		//    cfg - (optional, defaults to underscored flag) the name of the config file option
 		field := typ.Field(i)
+
+		// Recursively resolve embedded types.
+		if field.Anonymous {
+			if fieldVal := reflect.Indirect(val).FieldByName(field.Name); !fieldVal.IsNil() {
```

Comment:
Good point!  I'll add the ptr detection tomorrow

> On May 10, 2016, at 9:51 PM, Matt Reiferson <notifications@github.com> wrote:
> 
> In options.go:
> 
> > @@ -41,6 +41,14 @@ func Resolve(options interface{}, flagSet *flag.FlagSet, cfg map[string]interfac
> >  		//    deprecated - (optional) the name of the deprecated command line flag
> >  		//    cfg - (optional, defaults to underscored flag) the name of the config file option
> >  		field := typ.Field(i)
> > +
> > +		// Recursively resolve embedded types.
> > +		if field.Anonymous {
> > +			if fieldVal := reflect.Indirect(val).FieldByName(field.Name); !fieldVal.IsNil() {
> Does the Indirect assume that the embedded type is a pointer? Does this work if it isn't?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>